Unable to update application due to duplicated access token claims

### Description

After migrating to IS-7.2, when trying to update an OIDC application via the management console (e.g. access URL, public client, etc.), the update fails and the following error can be found in the logs:
```
java.sql.BatchUpdateException: Batch entry 9 INSERT INTO IDN_OAUTH2_TOKEN_CLAIMS (APP_ID, CLAIM_URI) VALUES (243,'street_address')  was aborted: ERROR: duplicate key value violates unique constraint "token_claims_constraint"  Detail: Key (app_id, claim_uri)=(243, street_address) already exists.  Call getNextException to see other errors in the batch.
  at org.postgresql.jdbc.BatchResultHandler.handleError(BatchResultHandler.java:154)
  at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2269)
  at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:511)
  at org.postgresql.jdbc.PgStatement.internalExecuteBatch(PgStatement.java:851)
  at org.postgresql.jdbc.PgStatement.executeBatch(PgStatement.java:874)
  at org.postgresql.jdbc.PgPreparedStatement.executeBatch(PgPreparedStatement.java:1569)
  at jdk.internal.reflect.GeneratedMethodAccessor303.invoke(Unknown Source)
  at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.base/java.lang.reflect.Method.invoke(Method.java:566)
  at org.apache.tomcat.jdbc.pool.StatementFacade$StatementProxy.invoke(StatementFacade.java:118)
  at com.sun.proxy.$Proxy117.executeBatch(Unknown Source)
  at org.wso2.carbon.identity.oauth.dao.OAuthAppDAO.addAccessTokenClaims(OAuthAppDAO.java:1655)
  at org.wso2.carbon.identity.oauth.dao.OAuthAppDAO.updateAccessTokenClaims(OAuthAppDAO.java:1631)
  at org.wso2.carbon.identity.oauth.dao.OAuthAppDAO.updateConsumerApplication(OAuthAppDAO.java:836)
  at org.wso2.carbon.identity.oauth.OAuthAdminServiceImpl.updateConsumerApplication(OAuthAdminServiceImpl.java:1031)
  at org.wso2.carbon.identity.oauth.OauthInboundAuthConfigHandler.handleConfigUpdate(OauthInboundAuthConfigHandler.java:157)
  at org.wso2.carbon.identity.application.mgt.ApplicationManagementServiceImpl.updateApplicationByResourceId(ApplicationManagementServiceImpl.java:2879)
  at org.wso2.carbon.identity.api.server.application.management.v1.core.ServerApplicationManagementService.updateServiceProvider(ServerApplicationManagementService.java:2155)
  at org.wso2.carbon.identity.api.server.application.management.v1.core.ServerApplicationManagementService.putApplicationInbound(ServerApplicationManagementService.java:2104)
  at org.wso2.carbon.identity.api.server.application.management.v1.core.ServerApplicationManagementService.putInboundOAuthConfiguration(ServerApplicationManagementService.java:1292)
  at org.wso2.carbon.identity.api.server.application.management.v1.impl.ApplicationsApiServiceImpl.updateInboundOAuthConfiguration(ApplicationsApiServiceImpl.java:464)
  at org.wso2.carbon.identity.api.server.application.management.v1.ApplicationsApi.updateInboundOAuthConfiguration(ApplicationsApi.java:1434)
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.base/java.lang.reflect.Method.invoke(Method.java:566)
  at org.apache.cxf.service.invoker.AbstractInvoker.performInvocation(AbstractInvoker.java:179)
  at org.apache.cxf.service.invoker.AbstractInvoker.invoke(AbstractInvoker.java:96)
  at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:201)
  at org.apache.cxf.jaxrs.JAXRSInvoker.invoke(JAXRSInvoker.java:104)
  at org.apache.cxf.interceptor.ServiceInvokerInterceptor$1.run(ServiceInvokerInterceptor.java:59)
  at org.apache.cxf.interceptor.ServiceInvokerInterceptor.handleMessage(ServiceInvokerInterceptor.java:96)
  at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:307)
  at org.apache.cxf.transport.ChainInitiationObserver.onMessage(ChainInitiationObserver.java:121)
  at org.apache.cxf.transport.http.AbstractHTTPDestination.invoke(AbstractHTTPDestination.java:267)
  at org.apache.cxf.transport.servlet.ServletController.invokeDestination(ServletController.java:234)
  at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:208)
  at org.apache.cxf.transport.servlet.ServletController.invoke(ServletController.java:160)
  at org.apache.cxf.transport.servlet.CXFNonSpringServlet.invoke(CXFNonSpringServlet.java:225)
  at org.apache.cxf.transport.servlet.AbstractHTTPServlet.handleRequest(AbstractHTTPServlet.java:304)
  at org.apache.cxf.transport.servlet.AbstractHTTPServlet.doPut(AbstractHTTPServlet.java:234)
  at javax.servlet.http.HttpServlet.service(HttpServlet.java:558)
  at org.apache.cxf.transport.servlet.AbstractHTTPServlet.service(AbstractHTTPServlet.java:279)
  at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:199)
  at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:144)
  at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
  at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:168)
  at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:144)
  at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:168)
  at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
  at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:482)
  at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:130)
  at org.wso2.carbon.identity.context.rewrite.valve.TenantContextRewriteValve.invoke(TenantContextRewriteValve.java:145)
  at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
  at org.wso2.carbon.identity.context.rewrite.valve.OrganizationContextRewriteValve.invoke(OrganizationContextRewriteValve.java:123)
  at org.wso2.carbon.tomcat.ext.valves.SameSiteCookieValve.invoke(SameSiteCookieValve.java:38)
  at org.wso2.carbon.identity.cors.valve.CORSValve.invoke(CORSValve.java:83)
  at org.wso2.carbon.identity.authz.valve.AuthorizationValve.invoke(AuthorizationValve.java:181)
  at org.wso2.carbon.identity.auth.valve.AuthenticationValve.invoke(AuthenticationValve.java:156)
  at org.wso2.carbon.tomcat.ext.valves.CompositeValve.continueInvocation(CompositeValve.java:110)
  at org.wso2.carbon.tomcat.ext.valves.TomcatValveContainer.invokeValves(TomcatValveContainer.java:49)
  at org.wso2.carbon.tomcat.ext.valves.CompositeValve.invoke(CompositeValve.java:71)
  at org.wso2.carbon.tomcat.ext.valves.CarbonStuckThreadDetectionValve.invoke(CarbonStuckThreadDetectionValve.java:150)
  at org.wso2.carbon.extension.identity.x509Certificate.valve.X509CertificateAuthenticationValve.invoke(X509CertificateAuthenticationValve.java:59)
  at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:660)
  at org.wso2.carbon.identity.core.context.valve.IdentityContextCreatorValve.invoke(IdentityContextCreatorValve.java:54)
  at org.wso2.carbon.tomcat.ext.valves.CarbonContextCreatorValve.invoke(CarbonContextCreatorValve.java:63)
  at org.wso2.carbon.tomcat.ext.valves.RequestEncodingValve.invoke(RequestEncodingValve.java:49)
  at org.wso2.carbon.tomcat.ext.valves.RequestCorrelationIdValve.invoke(RequestCorrelationIdValve.java:137)
  at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:761)
  at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
  at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:346)
  at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:396)
  at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
  at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:937)
  at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1793)
  at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
  at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1190)
  at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
  at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
  at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "token_claims_constraint"  Detail: Key (app_id, claim_uri)=(243, street_address) already exists.
  at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2533)
  at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2268)
  ... 79 more
```
The above error points to a claim that is configured more than once in the user attributes (e.g. `stret_address`) and this leads to adding this attribute more than once to the list of access token claims, which could be confirmed if the token type is temporarily changed to `JWT` to display the list of access token claims.

### Proposed Solution
Validate the selected user attributes to avoid duplicated claims.

### Steps to Reproduce

1. Create an application.
2. Manually change the VERSION column value of the respective application entry in SP_APP table to `v1.0.0`.
3. Update the application with following API request to add the same claim twice:
```
curl --location --request PATCH 'https://localhost:9443/api/server/v1/applications/<app_id>' \
--header 'Authorization: Basic YWRtaW46YWRtaW4=' \
--header 'Accept: application/json' \
--header 'Content-Type: application/json' \
--data '{
    "claimConfiguration": {
        "dialect": "LOCAL",
        "requestedClaims": [
            {
                "claim": { "uri": "http://wso2.org/claims/streetaddress" },
                "mandatory": false
            },
            {
                "claim": { "uri": "http://wso2.org/claims/streetaddress" },
                "mandatory": false
            },
            {
                "claim": { "uri": "http://wso2.org/claims/givenname" },
                "mandatory": false
            }
        ],
        "role": {
            "claim": { "uri": "" },
            "includeUserDomain": false
        },
        "subject": {
            "claim": { "uri": "http://wso2.org/claims/userid" },
            "includeTenantDomain": false,
            "includeUserDomain": false,
            "mappedLocalSubjectMandatory": false,
            "useMappedLocalSubject": false
        }
    }
}'
```
4. Go to the application configuration page and switch the token type option to JWT. Then you will see the duplicated entries.

### Please select the area issue is related to

API Access Mgt & Authorization

### Version

7.2.0

### Environment Details (with versions)

DB: Any database vendor (PostgreSQL used in this case)

### Developer Checklist

- [ ] [Behavioural Change] Does this change introduce a behavioral change to the product?
- [ ]  ↳ Approved by team lead
- [ ]  ↳ Label `impact/behavioral-change` added
- [ ] [Migration Impact] Does this change have a migration impact?
- [ ]  ↳ Migration label added (e.g., `7.2.0-migration`)
- [ ]  ↳ Migration issues created and linked
- [ ] [New Configuration] Does this change introduce a new configuration?
- [ ]  ↳ Label `config` added
- [ ]  ↳ Configuration is properly documented

--- a/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java
+++ b/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java
@@ -104,6 +104,7 @@ import java.util.Arrays;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Iterator;
+import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
@@ -3246,7 +3247,7 @@ public class OAuthAdminServiceImpl {
         ApplicationManagementService applicationMgtService = OAuth2ServiceComponentHolder
                 .getApplicationMgtService();
         try {
-            List<String> jwtAccessTokenClaims = new ArrayList<>();
+            Set<String> jwtAccessTokenClaims = new LinkedHashSet<>();
             ServiceProvider serviceProvider = applicationMgtService.getServiceProvider(oauthApp.getApplicationName(),
                     tenantDomain);
             if (serviceProvider != null) {

